### ISSUE TITLE
Job Board MVP

### ISSUE THREAD

**Issue by alysivji** [chicagopython/chipy.org#270]
This ticket represents the feature -- will need to write stories and connect them to this epic.

----

### Background

ChiPy has a [Jobs referral](https://www.chipy.org/pages/referrals/) program. There is also a #jobs channel in our Slack workspace.

### Proposed Feature

A job board with open positions that we can host on ChiPy.org would be a great compliment to our jobs program. It would benefit to our community in many ways:
- help members find new opportunities
- help sponsors find new employees
- fund activities and events to futher ChiPy's mission

### Minimum Requirements

- have a page showing a list of jobs -- high level overview
- can click into each job to get additional details about each job
- anybody can submit a job, only authorized users can approve jobs to appear
  - submitters have to accept the guidelines of ChiPy's [Jobs referral](https://www.chipy.org/pages/referrals/) policy
- important fields: `company`, `title`, `description` (accept markdown), `contact` (`name`, `email`, `title`), `link` (to company page)
  - definitely missed somethings here
- Are you an external recruiter? checkbox
- Is your organization interested in hosting a meetup? checkbox
- `created_at` and `updated_at` columns in all tables (if we don't already do that)

### Nice to haves

- "featured jobs" -- have a way for sponsors job
- search; good feature to add once we have more jobs 
- periodically post jobs from website to Twitter, LinkedIn, or Slack
  - it would be great if this jobs board is the only way people can promote jobs on our Slack

### Open Questions

- what are the different states a job posting can have?
- how long should each job posting be shown for?
- should we limit the number of job posting from a single organization?
- should we allow external recruiters to post jobs?

**Comment by alysivji** [comment]
@eevelweezel do you have any thoughts on this?

**Comment by eevelweezel** [comment]
I think we maybe should have posts auto-expire (say, 60 days?), and give submitters some mechanism to request removal of their post.  For sponsors, maybe we could provide a way to override this, and allow them to keep their post up for longer (maybe this ties into "featured jobs")? As for external recruiters, I'm fine with allowing them, provided the they agree to follow our guidelines AND provide the name of the company.

I'd say let's keep it to one active post per position.  While I like the idea of locking down the Slack channel to only posts from the jobs board, that wouldn't necessarily stop other forms of abuse, such as staffing firms outside of Chicago contacting employers about remote workers.  If we did lock down Slack, would members still be able to post their resumes?

**Comment by elmq0022** [comment]
I agree with the auto expire feature.  We could definitely make this
configurable, but 60 days sounds like a reasonable default.  I feel like a
lot of this has some overlap with how we handle Topics (approval by an
admin), and how we handle the RSVPs.  It would be good to give job posters
a way to remove their post once the position is filled as well.

**Comment by alysivji** [comment]
> I'd say let's keep it to one active post per position. While I like the idea of locking down the Slack channel to only posts from the jobs board, that wouldn't necessarily stop other forms of abuse, such as staffing firms outside of Chicago contacting employers about remote workers. If we did lock down Slack, would members still be able to post their resumes?

Locking down Slack would be a bit of a manual process. Our jobs channel doesn't get the traffic of say Chicago Tech Slack's jobs channel. I think we can have admins moderate until a better path forward is found.

**Comment by alysivji** [comment]
We had one of our sponsors try out the Jobs form and got some great feedback. For the next round of the MVP, we need the following:

- required field for "Company Website"
- required field for "How to Apply"
  - i would image folks will put URLs and instructions with email / subject requirements
- change the external recruiter question to: "Do you work for a recruiting agency?" or "Is this posting from a recruiting agency"
  - @eevelweezel do you have thoughts on wording?

----

We should also make the Base URL of the jobs board: chipy.org/job-board
 - think it's okay if we show the list view there
 - should also add a link on this page to "Submit Job" that takes us to the create view

----

Can we move the highlighted area all the way to the bottom of the page. It doesn't look great right now
![image](https://user-images.githubusercontent.com/4369343/81130430-59abce80-8f0d-11ea-8b98-29145705a747.png)

Can we have labels and form elements side by side?
![image](https://user-images.githubusercontent.com/4369343/81130492-86f87c80-8f0d-11ea-8ee7-52d48e6da9bc.png)

**Comment by eevelweezel** [comment]
I'm fine with "Is this posting from a recruiting agency?"

**Comment by sophm256** [comment]
@alysivji Would you like there to be a field regarding the location?

'Location' dropdown field that has the following choices: 'Chicago', 'Chicago and Remote' , 'Remote Only'

**Comment by eevelweezel** [comment]
I would like this, including a mention that we are a local user group.

**Comment by sophm256** [comment]
@eevelweezel  Great, I will include this field. How does this wording next to the field sound? 

`"ChiPy is a locally based group. The job position must not move the candidate out from Chicago. Working remotely or commuting is acceptable."`

**Comment by sophm256** [comment]
TASK 1
On the "after submit job post" page:

1. Instead of having the successful submit job message showing up hardcoded html template, it'll use Django's message system to have a dismissible pop-up.

2. On the "after submit job post" page, there'll be a list of to links of all the job posts the user has submitted.

For each job post submitted, there'll be a link to:

-Edit a job post (once it's submitted and before it's approved). Once the job post has been approved, there will not be an option to edit it, but to view it.

-An option to delete the job post once it's submitted. The user can delete prior to it being approved or after it's approved. (This will not be a soft delete but an actual delete of the post from the database.)

-There'll be a way for the user to specify how long they want the post to be shown for (60 days or less)
..................................................................................................................................
TASK 2
-When a user logins to ChiPy, there's a drop down menu that currently has the items "MY TALKS", "ACCOUNT", "+ ACCOUNT". Add a link called, "JOB POSTS" to this drop down menu. Have that link go to "after submit job"  post page.

....................................................................................................................................
TASK 3
-When a user submits a job post, send an email to the user saying that their post has been successfully submitted. 
```(e.g. 'Your job post for "Web Developer, Acme Company" has been successfully submitted. An admin will need to approve your post before it can be displayed on the job board. To see the status of your job post or to edit/delete it, go to <a href=link to after submit job post>your job post page</a>)```
......................................................................................................................................
TASK 4
-When an admin approves the job post, send an email to the user saying their post has been approved.
``` (e.g. 'Your job post for "Web Developer, Acme Company" has been approved. To see your job post, go to <a href=link to after submit job post>your job post page</a>')```

........................................................................................................................................
TASK 5.
-I am also thinking of changing the look of the listview that has the job posts.  Instead of being  

- **Web Developer at Acme Company, Posted Today**

it looks more like:

```
Web Developer        Posted Today
Acme Company
Chicago                    
```
The list items will not have bullets and will be separated by a grey line underneath each item.

**Comment by sophm256** [comment]
Also, for the list view of the job posts, do we want the user to scroll forever down or do we want to paginate? If we paginate, how many posts should be listed per page so that there's not a big gap if we float or sticky the footer to bottom of the page? Won't this be an issue for different screen sizes?

**Comment by elmq0022** [comment]
I suggest pagination with 10 to 20 post per page.  You can probably wait until we have enough postings for this to matter.

**Comment by sophm256** [comment]
@alysivji, @eevelweezel, @raymondberg, @elmq0022 
 
I am also intending to make the following changes for version two of the job board. Please let me know what you think about them.

1. In a previous comment, I said I'd be adding a 'location' field that would be a drop down menu with the following choices: 'Chicago', 'Chicago and Remote', and 'Remote Only'. **Now I will be adding a 4th choice 'Chicago and Temporarily Remote'.**

........................................................................................................................................................................................

2. I'll be adding a 'Job Type' field with a drop down menu of the following choices: 'Full Time', 'Part-Time', 'Contract', 'Paid Internship', 'Paid Apprenticeship', 'Other'. **Should people be allowed to post things that are unpaid, such as 'Volunteer', 'Unpaid Internship', 'Unpaid Apprenticeship'? What do you think of the categories I have listed so far?**

.........................................................................................................................................................................................

3. When an admin looks at a submitted job post and rejects the post, an email will be sent out to the user.  
  
3a) **Should the status be called 'denied' when an admin rejects the post?** Is there a nicer way to say this, as I'll be showing the 'status' of the job post to the user at chipy.org? **How about 'not approved'?**

 3b) **Should the rejection email be a general email** like 'Unfortunately, your job post that you submitted for Web Developer at Acme' has not been approved. Please contact <chipy's email> if you have any questions.' 

 3c) **Or would you like admins to have the ability to give specific reasons for the rejection?** I can make a 'Reason for Denial' field in the JobPost model that an admin can fill out. That way when the rejection email is sent, the specific reason for the rejection can be sent.  Also, the reason for the rejection can be seen when the user checks on the status of the job post at chipy.org. 

For example, specific reasons for rejections can be 'The job post is not approved because in the description field, the job posts says that the job is based outside of Chicago' or 'The job post is not approved for posting, because unprofessional language was used.'

**Comment by alysivji** [comment]
I think all these ideas are great, but we're still not done with the MVP. Let's get something out there and then we can discuss next steps.

**Comment by sophm256** [comment]
@alysivji Thanks for commenting. Was wondering, what do you consider to be the MVP?

**Comment by alysivji** [comment]
MVP for me is a job board our community can use. The feature is almost ready to release, but there [some outstanding items](https://github.com/chicagopython/chipy.org/issues/270#issuecomment-624396508) to complete. Would love to incorporate the feedback we got into the next iteration.

Once that is done, we can cut a release and share it with the community! I'm sure we will get additional feedback and we can iterate accordingly.

**Comment by raymondberg** [comment]
I'm catching up for my own understanding. It was a little dense on this issue for me to follow, as such I'm re-updating the list from [@alysivji outline](https://github.com/chicagopython/chipy.org/issues/270#issuecomment-624396508) and [breaking out into cards](https://github.com/chicagopython/chipy.org/projects/2) where not Done:

 * required field for "Company Website" _Done_
 * required field for "How to Apply" https://github.com/chicagopython/chipy.org/issues/301
 * change the external recruiter question to: "Do you work for a recruiting agency?" or "Is this posting from a recruiting agency"   #300
 * We should also make the Base URL of the jobs board: chipy.org/job-board. _Done_
   * think it's okay if we show the list view there _Done_ 
 * should also add a link on this page to "Submit Job" that takes us to the create view _Done_
 * Implement a min-height requirement for the body of the Job page so the footer doesn't show up near the header #302
 * Can we have labels and form elements side by side? _Done_
 * Location Fields include (Chicago, Chicago and Remote, Remote Only, Chicago and Temporarily Remote) #303 
 * Job Type Field includes (Full-Time, Contract to Hire Full Time, Part Time, Paid Internship, Paid Apprenticeship) #304

So it sounds like if @sophm256 can tackle the cards identified above, we are ready for MVP release review?

(BTW, this looks awesome!)

**Comment by alysivji** [comment]
With the MVP done, closing this in favour of #312 which will track the progress of issues related to the ChiPy Job Board.
